source/opt/def_use_manager.h

```diff
@@ -53,13 +54,41 @@ inline bool operator<(const Use& lhs, const Use& rhs) {
   return lhs.operand_index < rhs.operand_index;
 }
 
-using UseList = std::list<Use>;
+using UserEntry = std::pair<ir::Instruction*, ir::Instruction*>;
+
+struct less_than_id_ptr {
+  bool operator()(const UserEntry& lhs, const UserEntry& rhs) const {
+    if (!lhs.first && rhs.first)
+      return true;
+    if (lhs.first && !rhs.first)
+      return false;
+
+    // If lhs.first and rhs.first are both null, fall through to checking the
```

Comment:
Feels like this comment belongs after the "if" construct that starts on line 68, i.e. move to  around line 74.